internal/jimm/juju/migrationtarget.go

```diff
@@ -164,6 +165,22 @@ func (j *JujuManager) Prechecks(ctx context.Context, user *openfga.User, model m
 		return errors.E(op, fmt.Errorf("failed to modify migration info: %w", err))
 	}
 
+	_, err = j.Database.FindRegionByCloudName(ctx, model.ModelDescription.CloudCredential().Cloud(), model.ModelDescription.CloudRegion())
+	if err != nil {
+		return errors.E(op, fmt.Errorf("failed to find region for cloud %q: %w", model.ModelDescription.CloudCredential().Cloud(), err))
+	}
+
+	cloudCredential := &dbmodel.CloudCredential{
+		CloudName:         model.ModelDescription.CloudCredential().Cloud(),
+		OwnerIdentityName: model.ModelDescription.Owner().Id(),
+		Name:              model.ModelDescription.CloudCredential().Name(),
+	}
+
+	err = j.Database.GetCloudCredential(ctx, cloudCredential)
```

Comment:
Can merge line 173 here, it doesn't need an assigned variable as it isn't going to be reused. I'd prob godoc too explaining we're checking the cloud credential exists.